ghc-9.8.3: testsuite: Ambiguous occurrence ‘Unit’.

For Stackage Nightly ghc-9.8.3:

```
       Building test suite 'test' for generic-arbitrary-1.0.1..                                                                     
       [5 of 6] Compiling RecursiveTest                                                                                             
                                                                                                                                    
       /home/curators/work/unpack-dir/unpacked/generic-arbitrary-1.0.1-38783a7c474443404b797d9acbf1dfe3fd9afdd193eea4c42e26d9d82b2b3
c39/test/RecursiveTest.hs:20:44: error: [GHC-87543]                                                                                 
           Ambiguous occurrence ‘Unit’.                                                                                             
           It could refer to                                                                                                        
              either ‘Control.DeepSeq.Unit’,                                                                                        
                     imported from ‘Control.DeepSeq’ at test/RecursiveTest.hs:11:1-32,                                              
                  or ‘RecursiveTest.Unit’, defined at test/RecursiveTest.hs:16:1.                                                   
          |                                                                                                                         
       20 |   deriving Arbitrary via (GenericArbitrary Unit)      
          |                                            ^^^^                                                                         
                                                                                                                                    
       /home/curators/work/unpack-dir/unpacked/generic-arbitrary-1.0.1-38783a7c474443404b797d9acbf1dfe3fd9afdd193eea4c42e26d9d82b2b3
c39/test/RecursiveTest.hs:27:19: error: [GHC-87543]                                                                                 
           Ambiguous occurrence ‘Unit’.                           
           It could refer to                                                                                                        
              either ‘Control.DeepSeq.Unit’,                                                                                        
                     imported from ‘Control.DeepSeq’ at test/RecursiveTest.hs:11:1-32,                                              
                  or ‘RecursiveTest.Unit’, defined at test/RecursiveTest.hs:16:1.                                                   
          |                                                       
       27 | prop_unitTotal :: Unit -> Property                                                                                      
          |                   ^^^^                                                                                                  
                                                                  
       /home/curators/work/unpack-dir/unpacked/generic-arbitrary-1.0.1-38783a7c474443404b797d9acbf1dfe3fd9afdd193eea4c42e26d9d82b2b3
c39/test/RecursiveTest.hs:59:17: error: [GHC-87543]               
           Ambiguous occurrence ‘Unit’.                                                                                             
           It could refer to                                      
              either ‘Control.DeepSeq.Unit’,                                                                                        
                     imported from ‘Control.DeepSeq’ at test/RecursiveTest.hs:11:1-32,
:
:
```

--- a/test/RecursiveTest.hs
+++ b/test/RecursiveTest.hs
@@ -8,7 +8,7 @@
 
 module RecursiveTest where
 
-import           Control.DeepSeq
+import           Control.DeepSeq hiding (Unit)
 import           GHC.Generics                      (Generic)
 import           Test.QuickCheck
 import           Test.QuickCheck.Arbitrary.Generic

